### PR TITLE
Use cookies

### LINKED ISSUE #43: Use cookies to allow different usersessions on different browsers
Only one user can stay logged into the system at the moment

if you register and login with one account, other (previous) users cannot log in - it shows an error. But shopping cart is updated for that user every time you log in.

### PR DESCRIPTION
**Pull request by thanujasiva**

### CODE REVIEW

**Review by shrimeichock — APPROVED** [review]

**Review by maishabd23 — APPROVED** [review]

File: src/main/java/bookstore/users/UserController.java [hunk 1]
```diff
@@ -195,32 +187,68 @@ public String handleUserLogin(@ModelAttribute BookUser formUser, Model model) {
         }
     }
 
-    public BookUser getLoggedInUser(){
-
-        List<Usersession> usersessions = (List<Usersession>) usersessionRepository.findAll();
+    public BookUser getLoggedInUser(Cookie[] cookies){
```
Comment (thanujasiva) [inline src/main/java/bookstore/users/UserController.java:203]: `getLoggedInUser` now uses the request cookies to return the corresponding BookUser.

This method calls the new static method `retreieveCookie`.

File: src/main/java/bookstore/users/Usersession.java
Comment (thanujasiva) [inline src/main/java/bookstore/users/Usersession.java:1]: The `Usersession` class is no longer needed

File: src/main/java/bookstore/inventory/CheckoutController.java [hunk 2]
```diff
@@ -53,15 +55,16 @@ public CheckoutController(AuthorRepository authorRepo, BookRepository bookRepo,
      */
     @GetMapping("/listAvailableBooks")
     public String listAvailableBooks
-    (@RequestParam(name = "searchValue", required = false, defaultValue = "") String searchValue,
+    (HttpServletRequest request, HttpServletResponse response,
```
Comment (thanujasiva) [inline src/main/java/bookstore/inventory/CheckoutController.java:58]: Methods that require user access now need to pass in the request/response parameters

File: src/main/java/bookstore/inventory/CheckoutController.java [hunk 3]
```diff
@@ -329,46 +338,29 @@ public String removeFromCart (@RequestParam(name = "selectedItems", required = f
...
-        return "home";    
-}
-
-    /**
-     * Method to retrieve shopping cart for user
-     * @return the shopping cart
-     * @author Maisha Abdullah
-     * @author Thanuja Sivaananthan
-     */
-    private ShoppingCart getOrCreateShoppingCart(){
```
Comment (thanujasiva) [inline src/main/java/bookstore/inventory/CheckoutController.java:347]: `getOrCreateShoppingCart` is no longer needed. Instead, we first check if the loggedInUser exists, then we can call `getShoppingCart` on that user.

File: src/test/java/bookstore/inventory/CheckoutControllerTest.java [hunk 4]
```diff
@@ -147,11 +150,17 @@ void testViewCartWithAccess(){
...
+        HttpServletRequest request = new MockHttpServletRequest();
+        HttpServletResponse response = new MockHttpServletResponse();
         ShoppingCart shoppingCart = new ShoppingCart(inventory);
         shoppingCart.addToCart(book1, 3);
         shoppingCart.addToCart(book2, 1);
 
-        String view = controller.viewCart(model);
+        BookUser bookUser = new BookUser("testUser", "password123");
+        bookUser.setShoppingCart(shoppingCart);
+        when(userController.getLoggedInUser(request.getCookies())).thenReturn(bookUser);
```
Comment (thanujasiva) [inline src/test/java/bookstore/inventory/CheckoutControllerTest.java:161]: Controller tests that require user access need this Mock

**Review by sabahsamwatin — APPROVED** [review]

**Pull request merged**